internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/controller/kafka/consumer/AdminConsumptionTaskIntegrationTest.java

```diff
@@ -101,6 +109,103 @@ public void testSkipMessageEndToEnd() throws ExecutionException, InterruptedExce
     }
   }
 
+  @Test(timeOut = TIMEOUT)
+  public void testParallelAdminExecutionTasks() throws IOException {
+    try (ZkServerWrapper zkServer = ServiceFactory.getZkServer();
+        PubSubBrokerWrapper pubSubBrokerWrapper = ServiceFactory.getPubSubBroker(
+            new PubSubBrokerConfigs.Builder().setZkWrapper(zkServer).setRegionName(STANDALONE_REGION_NAME).build());
+        TopicManager topicManager =
+            IntegrationTestPushUtils
+                .getTopicManagerRepo(
+                    PUBSUB_OPERATION_TIMEOUT_MS_DEFAULT_VALUE,
+                    100,
+                    0l,
+                    pubSubBrokerWrapper,
+                    pubSubTopicRepository)
+                .getLocalTopicManager()) {
+      PubSubTopic adminTopic = pubSubTopicRepository.getTopic(AdminTopicUtils.getTopicNameFromClusterName(clusterName));
+      topicManager.createTopic(adminTopic, 1, 1, true);
+      String storeName = "test-store";
+      int adminConsumptionMaxWorkerPoolSize = 3;
+      extraProperties.put(ADMIN_CONSUMPTION_MAX_WORKER_THREAD_POOL_SIZE, adminConsumptionMaxWorkerPoolSize);
+      extraProperties.put(ADMIN_CONSUMPTION_CYCLE_TIMEOUT_MS, 3000);
+      try (
+          VeniceControllerWrapper controller = ServiceFactory.getVeniceController(
+              new VeniceControllerCreateOptions.Builder(clusterName, zkServer, pubSubBrokerWrapper)
+                  .regionName(STANDALONE_REGION_NAME)
+                  .extraProperties(extraProperties)
+                  .build());
+          PubSubProducerAdapterFactory pubSubProducerAdapterFactory =
+              pubSubBrokerWrapper.getPubSubClientsFactory().getProducerAdapterFactory();
+          VeniceWriter<byte[], byte[], byte[]> writer =
+              IntegrationTestPushUtils.getVeniceWriterFactory(pubSubBrokerWrapper, pubSubProducerAdapterFactory)
+                  .createVeniceWriter(new VeniceWriterOptions.Builder(adminTopic.getName()).build())) {
+        int executionId = 1;
+        byte[] goodMessage =
+            getStoreCreationMessage(clusterName, storeName, owner, keySchema, valueSchema, executionId);
+        writer.put(new byte[0], goodMessage, AdminOperationSerializer.LATEST_SCHEMA_ID_FOR_ADMIN_OPERATION);
+
+        // Spin up a thread to occupy the store write lock to simulate the blocking admin execution task thread.
+        Runnable infiniteLockOccupy = getRunnable(controller, storeName);
+        Thread infiniteLockThread = new Thread(infiniteLockOccupy, "infiniteLockOccupy: " + storeName);
+        infiniteLockThread.start();
+
+        for (int i = 0; i < adminConsumptionMaxWorkerPoolSize; i++) {
+          // Here we wait here to send every operation to let each consumer pool has at most one admin operation from
+          // this store.
+          Utils.sleep(5000);
```

Comment:
Here we create a store update request every 5 seconds > ADMIN_CONSUMPTION_CYCLE_TIMEOUT_MS(3 seconds), which will simulate a scenario:  first store's update will not finish due to not able to acquiring the store level lock. With previous implementation with checking if there is existing `AdminExecutionTask`, previous logic will try to create one new  `AdminExecutionTask` to exhaust the `ADMIN_CONSUMPTION_MAX_WORKER_THREAD_POOL_SIZE`, since each creation will be triggered by new polling of message of each store update. In short, 5 seconds waiting with 3 seconds timeout is try to achieve: all store update message will be in separate consumer poll and invoke a check if a store has an ongoing `AdminExecutionTask`.